validator/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/dadosjusbr/coletores/status"
+	"github.com/dadosjusbr/storage"
+	"github.com/frictionlessdata/tableschema-go/schema"
+	"github.com/frictionlessdata/tableschema-go/table"
+)
+
+type executionResult struct {
+	Pr storage.PackagingResult `json:"pr,omitempty"`
+	Cr storage.CrawlingResult  `json:"cr,omitempty"`
+}
+
+type dataExport struct {
+	csv   [][]string
+	dtpck map[string]interface{}
+}
+
+func main() {
+	var er executionResult
+	erIN, err := ioutil.ReadAll(os.Stdin)
+	if err != nil {
+		status.ExitFromError(status.NewError(4, fmt.Errorf("Error reading crawling result: %q", err)))
+	}
+	if err = json.Unmarshal(erIN, &er.Cr); err != nil {
+		status.ExitFromError(status.NewError(5, fmt.Errorf("Error unmarshaling crawling resul from STDIN: %q", err)))
+		os.Exit(1)
+	}
+	csvContent := mountCSV(er.Cr)
+	for i := range csvContent {
+		csvContent[i][8] = strings.TrimSpace(csvContent[i][8])
+	}
+	table := table.FromSlices(csvContent[0], csvContent[1:])
+	dtpckg := mountDtPckg(er.Cr)
+	sch := mountSchema(dtpckg)
+	var emp []employees
+	if err := sch.CastTable(table, &emp); err != nil {
+		fmt.Println(err)
+	}
+	var v dataExport
+	v.csv = csvContent
+	v.dtpck = dtpckg
+	fmt.Println(v)
+}
+
+//mountCSV receives a CR and returns header and content of a csv.
+func mountCSV(cr storage.CrawlingResult) [][]string {
+	var csvContent [][]string
+	headers := []string{"aid", "year", "month",
+		"reg", "name", "role", "type", "workplace", "active", "income_total", "wage",
+		"perks_total", "perks_food", "perks_transportation", "perks_preschool", "perks_health", "perks_birthaid", "perks_housingaid", "perks_subsistence", "perks_others",
```

Comment:
Morro de medo da complexidade incremental que a gente vem adicionando. Vamos fazer um pequeno exercício: e se a gente quiser adicionar uma informação a mais sobre o empregado, temos que?

1. Mudar a struct `Employee` no pacote storage
1. Mudar o schema.json nesse repositório
1. Mudar todos os coletores
1. Mudar a struct `employees` no pacote validator, ele tem que ter o nome do campo ou tableheader tem que bater com o que foi definido no schema.json
1. Mudar o esse `[]string` dos headers 
1. Se for um campo somatório (ex. Perk), lembrar que os totais são calculados nos coletores e também aqui no validador e alterar o validador

E isso não está documentado em nenhum lugar :sob: 

Estou considerando que os três primeiros são inevitáveis ou muito complicados de atacar. Uma vez que estamos sempre de olho em como melhorar o design e diminuir a complexidade do dadosjusbr, acho que podemos começar a agir e diminuir o débito técnico. Algumas sugestões:

- Não ter um slice de cabeçalhos hardcoded:

Notei que você só precisa desse slice hardcoded aqui para satisfazer o método `table.FromSlice`. 

Uma coisa importante notar é que o método `CastTable` usa [CastRow](https://github.com/frictionlessdata/tableschema-go/blob/master/schema/schema.go#L194) e esse último usa a posição do campo no esquema (ordem em que foi declarado) para achar a posição do slice de strings que representa a linha.  Acabei de refletir sobre essa implementação e ela é bem ruim. Se houver demanda, irei melhorar ela no futuro.

Ou seja, na realidade, você não precisa dos cabeçalhos. Dito isso, pode passar nil e adicionar um comentário explicando isso. 

- Não recalcular os somatórios no validador

Colocamos esse cálculo aqui pois percebemos que alguns coletores não estavam realizando essa conta. Ao invés de voltar e descobrir, a gente preferiu resolver por aqui. Foi importante para a apresentação, mas uma vez que vamos re-rodar os pipelines, podemos resolver o problema na fonte (nos coletores).

Para tal, temos que adicionar uma checagem para que os campos somatório seja sempre preenchido. Isso pode ser feito modificando-se o esquema adicionando uma [constraint](https://specs.frictionlessdata.io/table-schema/#constraints) `minimum: 0`. Isso é implementado [aqui](https://github.com/frictionlessdata/tableschema-go/blob/master/schema/number.go#L41). Temos que fazer isso para todos os campos que a gente julgar importantes que devem ser preenchidos pelos coletores.

- usar [gocsv](https://github.com/gocarina/gocsv) para gerar CSV do `CrawlingResult.Employees`

Usei recentemente em outro projeto e facilitou muito as coisas. Vamos ver se funciona.